nomic/data_operations.py

```diff
@@ -846,3 +846,125 @@ def remove(self, ids: List[str], tags: List[str], delete_all: bool = False) -> b
+        try:
+            # Run fetch_tiles first to guarantee existence of quad feather files
+            self._tb: pa.Table = self.projection._fetch_tiles()
+            sidecars, _ = self._download_data()
+            self._fetch_tiles_with_all_sidecars(sidecars)
+
+        except pa.lib.ArrowInvalid as e:
+            raise ValueError("Failed to fetch tiles for this map")
+
+    def _fetch_tiles_with_all_sidecars(self, additional_sidecars=None):
+        tbs = []
+        root = feather.read_table(self.projection.tile_destination / "0/0/0.feather")
+        try:
+            small_sidecars = set(
+                [v for k, v in json.loads(root.schema.metadata[b"sidecars"]).items()]
+            )
+        except KeyError:
+            small_sidecars = []
+        for path in self.projection._tiles_in_order():
+            tb = pa.feather.read_table(path).select(["_id"])
+            for sidecar_file in small_sidecars:
+                carfile = pa.feather.read_table(
+                    path.parent / f"{path.stem}.{sidecar_file}.feather"
+                )
+                for col in carfile.column_names:
+                    tb = tb.append_column(col, carfile[col])
+            for big_sidecar in additional_sidecars:
+                fname = base64.urlsafe_b64encode(big_sidecar.encode("utf-8")).decode(
+                    "utf-8"
+                )
+                carfile = pa.feather.read_table(
+                    path.parent / f"{path.stem}.{fname}.feather", memory_map=True
+                )
+                for col in carfile.column_names:
+                    tb = tb.append_column(col, carfile[col])
+            tb = tb.drop(["_id"])
+            tbs.append(tb)
+        self._tb = pa.concat_tables(tbs)
+
+        return self._tb
+
+    def _download_data(self):
+        """
+        Downloads the feather tree for large sidecar columns.
+        """
+        self.projection.tile_destination.mkdir(parents=True, exist_ok=True)
+        root = f"{self.project.atlas_api_path}/v1/project/public/{self.project.id}/index/projection/{self.projection.id}/quadtree/"
+
+        all_quads = list(self.projection._tiles_in_order(coords_only=True))
+        sidecars = [
+            field
+            for field in self.project.project_fields
+            if field not in self._tb.column_names and field != "_embeddings"
+        ]
+        returned_files = []
+
+        for quad in tqdm(all_quads):
+            quad_files = []
+            for sidecar in sidecars:
+                quad_str = "/".join([str(q) for q in quad])
+                encoded_colname = base64.urlsafe_b64encode(
+                    sidecar.encode("utf-8")
+                ).decode("utf-8")
+                filename = quad_str + "." + encoded_colname + ".feather"
+                path = self.projection.tile_destination / filename
+                quad_files.append(path)
+
+                # WARNING: Potentially large data request here
```

Comment:
I'm not super-worried about this--it's at most 50,000 points times 100kb per point, which is... hmm like a few gigabytes. Yeah that's pretty big actually. Might want to switch to just writing the bytes as they come in. https://stackoverflow.com/questions/16694907/download-large-file-in-python-with-requests

Reply:
Makes sense. Is there any benefit to this reading in the bytes, reading it in with `feather.read_table` than writing it?